echopype/calibrate/calibrate_base.py

```diff
@@ -7,13 +13,158 @@
+                )
+                values = values.data
+                xi = np.column_stack(
+                    (
+                        interp_plat["latitude"].data,
+                        interp_plat["longitude"].data,
+                    )
+                )
+                interp = scipy.interpolate.griddata(
+                    points, values, xi, method=self.interp_method
+                )
+                result[var] = ("ping_time", interp)
+            env_params = xr.Dataset(
+                data_vars=result, coords={"ping_time": interp_plat["ping_time"]}
+            )
+        else:
+            min_max = {
+                dim: {"min": env_params[dim].min(), "max": env_params[dim].max()}
+                for dim in dims
+            }
+
+            extrap = env_params.interp(
+                {dim: echodata.platform[dim].data for dim in dims},
+                method=self.extrap_method,
+                # scipy interp uses "extrapolate" but scipy interpn uses None
+                kwargs={"fill_value": "extrapolate" if len(dims) == 1 else None},
+            )
+            extrap_unique_idx = {
+                dim: np.unique(extrap[dim], return_index=True)[1] for dim in dims
+            }
+            extrap = extrap.isel(**extrap_unique_idx)
+            interp = env_params.interp(
+                {dim: echodata.platform[dim].data for dim in dims},
+                method=self.interp_method,
+            )
+            interp_unique_idx = {
+                dim: np.unique(interp[dim], return_index=True)[1] for dim in dims
+            }
+            interp = interp.isel(**interp_unique_idx)
+
+            if self.extrap_method is not None:
+                less = extrap.sel(
+                    {
+                        dim: extrap[dim][extrap[dim] < min_max[dim]["min"]]
+                        for dim in dims
+                    }
+                )
+                middle = interp.sel(
+                    {
+                        dim: interp[dim][
+                            np.logical_and(
+                                interp[dim] >= min_max[dim]["min"],
+                                interp[dim] <= min_max[dim]["max"],
+                            )
+                        ]
+                        for dim in dims
+                    }
+                )
+                greater = extrap.sel(
+                    {
+                        dim: extrap[dim][extrap[dim] > min_max[dim]["max"]]
+                        for dim in dims
+                    }
+                )
+
+                # remove zero length dims (xarray does not support in combine_by_coords)
+                non_zero_dims = [
+                    ds
+                    for ds in (less, middle, greater)
+                    if all(dim_len > 0 for dim_len in ds.dims.values())
+                ]
+                env_params = xr.combine_by_coords(non_zero_dims)
```

Comment:
Zero length dimension -- is this when one of `less`, `middle`, `greater` is empty?
Could you put in a short comment to clarify?